api/lib/validation/attributeValidator.js

```diff
@@ -0,0 +1,230 @@
+  var attributeValueErrors = [];
+  var errorCount = 0; 
+  for(let i = 0; i < inputSpecificDates.length; i++) {
+    if(inputSpecificDates[i].instance_min_count >=
+      inputSpecificDates[i].instance_max_count) {
+      let error = createErrorResponse('specific_date.instance_min_count',
+      'specific_date.instance_min_count and specific_date.instance_max_count' + 
+      ' values are not compatible',inputSpecificDates[i],'instance_min_count ' + 
+      inputSpecificDates[i].instance_min_count + ' is higher or equal to ' + 
+      'instance_max_count ' + inputSpecificDates[i].instance_max_count + 
+      ' in specific_date :[' + i + ']');
+      attributeValueErrors[errorCount++] = error;
+    }
+    if(!compareDateTimes(inputSpecificDates[i].start_date_time,
+      inputSpecificDates[i].end_date_time)) {
+      let error = createErrorResponse('specific_date.start_date_time',
+      'specific_date.start_date_time and specific_date.end_date_time ' +
+      'values are not compatible',inputSpecificDates[i],'start_date_time ' + 
+      inputSpecificDates[i].start_date_time + ' is after end_date_time ' + 
+      inputSpecificDates[i].end_date_time + ' in specific_date :[' + i + ']');
+      attributeValueErrors[errorCount++] = error;
+    }
+  }
+  return attributeValueErrors;
+}
+var validateSpecificDateErrors = function(inputSpecificDates) {
+  var errors = [];
+  var attributeValueErrors = [];
+  var dateRangeErrors = [];
+  if(inputSpecificDates.length > 0) {
+    attributeValueErrors = validateSpecificDateAttributeValue(inputSpecificDates);
+    dateRangeErrors = validateSpecificDateRange(inputSpecificDates);
+    errors = attributeValueErrors.concat(dateRangeErrors);
+  }
+  return errors;
+}
+
+var validatePolicyJSONValues = function(policyJson) {
+  var errorCount = 0;
+  var errors = [];
+  if(policyJson.schedules) {
+    var inputRecurringSchedules = policyJson.schedules.recurring_schedule;
+    var inputSpecificDates = policyJson.schedules.specific_date;
+    if(policyJson.instance_min_count >= policyJson.instance_max_count) {
+      errors[errorCount++] = createErrorResponse('instance_min_count','instance_min_count and ' + 
+      'instance_max_count values are not compatible', policyJson,'instance_min_count ' + 
+      policyJson.instance_min_count + ' is higher or equal to instance_max_count ' + 
+      policyJson.instance_max_count + ' in policy_json');
+    }
+    var specificDateErrors = [];
+    var recurringScheduleErrors = [];
+    if(inputSpecificDates) {
+      specificDateErrors = validateSpecificDateErrors(inputSpecificDates);
+    }
+    if(inputRecurringSchedules) {
+      recurringScheduleErrors = validateRecurringRuleErrors(inputRecurringSchedules); 
+    }
+    errors = errors.concat(specificDateErrors,recurringScheduleErrors);
+  }
+  
+  return errors;
+}
+
+
+
+exports.validatePolicy = function(inputPolicy,callback) {
+  if(callback) {
+    var result = validatePolicyJSONValues(inputPolicy); 
```

Comment:
'result' could either be an empty array (means everything's fine) or an array of errors to be passed back via the callback.